### ISSUE TITLE
Provide a configuration property for configuring an external jOOQ settings file

### ISSUE THREAD

**Issue by MelleD** [spring-projects/spring-boot#38778]
With Spring Boot you are used to setting most settings via `customizer` or `properties`.

I would like the same for jOOQ, e.g. with the batch size

https://github.com/jOOQ/jOOQ/blob/main/jOOQ/src/main/java/org/jooq/conf/Settings.java#L423C46-L423C46

The jOOQ properties should be expanded to be able to overwrite some/all default settings.

**Comment by MelleD** [comment]
@wilkinsona thanks for the quick answer.

That makes perfect sense.

My thoughts were:
a) Start with a few properties and extend if when needed
b) Build it generically like property key = setting#method
c) Build a customizer that you can at least inject your own propeties and then set them manually

c) we are currently doing more or less, but its not really nice

The problem with xml is that you cannot use other properties (e.g. for reuse) or the other mechanism for overwriting via ENV or config server etc. does not work.

**Comment by wilkinsona** [comment]
We discussed this today and decided that we'd like to provide a configuration property for specifying an external `settings.xml` to use as well as a customizer to make programmatic configuration of the settings more straightforward.

**Comment by ahrytsiuk** [comment]
@MelleD @mhalbritter  there is already a customizer for jOOQ:
```java
org.springframework.boot.autoconfigure.jooq.DefaultConfigurationCustomizer
```

**Comment by mhalbritter** [comment]
You're right, thanks!

**Comment by mhalbritter** [comment]
I have some changes here: https://github.com/mhalbritter/spring-boot/tree/mh/38778-provide-a-configuration-property-for-configuring-an-external-jooq-settings-file

We have to load the `Settings` through JAXB ourselves (i found no way to get jOOQ to load the config - the code path for `-Dorg.jooq.settings` uses an `@Internal` marked class). That means JAXB has to be on the classpath for that feature to work.

Right now, it silently ignores `spring.jooq.config` if JAXB is not on the classpath (but it's visible in the configuration report). Should we fail explicitly if `spring.jooq.config` is set and JAXB isn't there?

**Comment by wilkinsona** [comment]
> Should we fail explicitly if spring.jooq.config is set and JAXB isn't there?

Yes, I think so. We could throw a custom exception and provide a failure analyser for it.
